ext/reflection/php_reflection.c

```diff
@@ -4938,7 +4938,25 @@ ZEND_METHOD(ReflectionClass, newInstance)
 		HashTable *named_params;
 
 		if (!(constructor->common.fn_flags & ZEND_ACC_PUBLIC)) {
-			zend_throw_exception_ex(reflection_exception_ptr, 0, "Access to non-public constructor of class %s", ZSTR_VAL(ce->name));
+			zend_string *message_suffix = ZSTR_EMPTY_ALLOC();
+
+			if (zend_attribute_get_nonpublic_suffix(
+				constructor->common.attributes,
+				&message_suffix) == FAILURE
+			) {
+				ZEND_ASSERT(EG(exception));
+				zval_ptr_dtor(return_value);
+				RETURN_THROWS();
+			}
+		
+			zend_throw_exception_ex(
+				reflection_exception_ptr,
+				0,
+				"Access to non-public constructor of class %s%s",
```

Comment:
Nit: I'd probably prefer if this error message was similar to what is used elsewhere:

`Cannot call %s constructor of class %s%s`

There's also another variant of the same error message which is ` Call to private ...`, so I think the best would be to choose only one.